src/components/Link/Link.stories.tsx

```diff
@@ -137,3 +138,24 @@ export const JumpLinkIconLeft: Story = {
     </Link>
   )
 };
+
+export const LinkWithReactRouterLink: Story = {
+  name: 'Link using React Router Link',
+  parameters: {
+    docs: {
+      description: {
+        story:
+          'See [React Router Link docs](https://reactrouter.com/en/main/components/link) for usage information'
+      }
+    }
+  },
+  render: () => (
+    <BrowserRouter>
+      <p>
+        <Link href='/#' isRouterLink>
+          Link using React Router Link
```

Comment:
Note: after writing out everything below I realized that the fact that it is a CFPB styled Link is important to the name, so you can ignore the rest of this comment 😬 

---

The naming feels a bit awkward.  Since it's ultimately returning a RouterLink it would be easier to simply call this `React Router Link`. 
 
That said, we've been adjusting naming of variants in other components (ex. Instead of "Checkboxes -> Large target area checkbox" using "Checkboxes -> Large target area" to leverage the context that implies that everything under the "Checkboxes" heading is a "checkbox") and the names for Link could probably use some improvement as well.  So this is probably fine as-is and can be re-evaluated in the next round of naming reviews.